dbt/parser/schemas.py

```diff
@@ -343,13 +361,21 @@ def parse_v2_yml(cls, original_file_path, test_yml, package_name,
             return
 
         for model in test_yml['models']:
+            if not isinstance(model, dict):
+                msg = (
+                    "Invalid test config given in {} near {} (expected a dict)"
+                ).format(original_file_path, model)
+                if dbt.flags.STRICT_MODE:
+                    dbt.exceptions.raise_compiler_error(msg, model)
+                dbt.utils.compiler_warning(model, msg)
+                continue
             try:
                 model = UnparsedNodeUpdate(**model)
-            except dbt.exceptions.ValidationException as exc:
+            except dbt.exceptions.JSONValidationException as exc:
                 # we don't want to fail the full run, but we do want to fail
                 # parsing this file
-                msg = "Invalid test config given in {} near {}: {}".format(
-                        original_file_path, model, exc
+                msg = "Invalid test config given in {}: {}".format(
+                        original_file_path, exc.errors_message
```

Comment:
I think the warning we show two lines below here is kind of confusing. Presently, dbt shows the following warning in this code path:
```
* Compilation warning while compiling model event:
* Invalid test config given in models/schema.yml: Additional properties are not allowed ('descriptin' was unexpected)
```

This is a test, not a mode, so I can imagine someone being confused when they see this warning. Can we just quickly change the `compiler_warning` in `utils.py` to accept a resource type?

Then we could do:
```
dbt.utils.compiler_warning(model.get('name', 'unknown'), msg, resource_type='test')
```